src/main/java/sdu/sem2/se17/domain/CreditManagementHandlerImpl.java

```diff
@@ -23,9 +23,11 @@
     private final ProductionHandler productionHandler;
     private final UserHandler userHandler;
 
-    public CreditManagementHandlerImpl(DataSource dataSource) {
-        creditHandler = new CreditHandlerImpl(dataSource);
+    public CreditManagementHandlerImpl() {
+        var dataSource = new DataSource(null, null, null);
```

Comment:
I don't think the constructors way of handling the datasource should be changed, as this removes the possibility of connecting to the database. Moreover, currently two constructors have the same signature, so the program cannot compile.